Product/Production/Services/DocumentQueryCore/src/test/java/gov/hhs/fha/nhinc/docquery/audit/transform/DocQueryDeferredResponseAuditTransformsTest.java

```diff
@@ -0,0 +1,289 @@
+            @Override
+            protected String getLocalHostAddress() {
+                return LOCAL_IP;
+            }
+
+            @Override
+            protected String getRemoteHostAddress(Properties webContextProperties) {
+                if (webContextProperties != null && !webContextProperties.isEmpty()
+                    && webContextProperties.getProperty(NhincConstants.REMOTE_HOST_ADDRESS) != null) {
+
+                    return webContextProperties.getProperty(NhincConstants.REMOTE_HOST_ADDRESS);
+                }
+                return AuditTransformsConstants.ACTIVE_PARTICIPANT_UNKNOWN_IP_ADDRESS;
+            }
+
+            @Override
+            protected String getWebServiceUrlFromRemoteObject(NhinTargetSystemType target, String serviceName) {
+                return REMOTE_OBJECT_URL;
+            }
+        };
+
+        AssertionType assertion = createAssertion();
+        LogEventRequestType auditResponse = transforms.transformResponseToAuditMsg(request, response, assertion, null,
+            NhincConstants.AUDIT_LOG_INBOUND_DIRECTION, NhincConstants.AUDIT_LOG_NHIN_INTERFACE, Boolean.FALSE,
+            webContextProperties, NhincConstants.DOC_QUERY_DEFERRED_RESULTS_SERVICE_NAME);
+
+        testGetEventIdentificationType(auditResponse, NhincConstants.DOC_QUERY_DEFERRED_RESULTS_SERVICE_NAME,
+            Boolean.FALSE);
+        testAuditSourceIdentification(auditResponse.getAuditMessage().getAuditSourceIdentification(), assertion);
+        testGetActiveParticipantSource(auditResponse, Boolean.FALSE, webContextProperties, LOCAL_IP);
+        testGetActiveParticipantDestination(auditResponse, Boolean.FALSE, webContextProperties, REMOTE_OBJECT_URL);
+        assertParticipantObjectIdentification(auditResponse, hcid);
+        assertEquals("AuditMessage.Response ServiceName mismatch", auditResponse.getEventType(),
+            NhincConstants.DOC_QUERY_DEFERRED_RESULTS_SERVICE_NAME);
+    }
+
+    private void assertParticipantObjectIdentification(LogEventRequestType auditRequest, String hcid) {
+        assertNotNull("auditRequest is null", auditRequest);
+        assertNotNull("AuditMessage is null", auditRequest.getAuditMessage());
+        assertNotNull("ParticipantObjectIdentification is null",
+            auditRequest.getAuditMessage().getParticipantObjectIdentification());
+        assertFalse("ParticipantObjectIdentification list is missing test values",
+            auditRequest.getAuditMessage().getParticipantObjectIdentification().size() < 1);
+
+        ParticipantObjectIdentificationType participantPatient = auditRequest.getAuditMessage()
+            .getParticipantObjectIdentification().get(0);
+
+        assertEquals("ParticipantPatient.ParticipantObjectID mismatch", "D123401^^^&1.1&ISO",
+            participantPatient.getParticipantObjectID());
+
+        // TODO: assertSame vs assertEquals consistency when returning constants
```

Comment:
Was this TODO completed?

Reply:
sorry that cut and paste from the old test
should i removed?